### PR TITLE
Backport patch https://github.com/brianmario/mysql2/pull/811

### PR DESCRIPTION
**Pull request by tamimkh**
Backport patch in order to resolve issues with using rack timeout and mysql connections being terminted while reading results

### CODE REVIEW

File: ext/mysql2/client.c [hunk 1]
```diff
@@ -595,20 +595,20 @@ static VALUE do_query(void *args) {
   return Qnil;
 }
 #else
-static VALUE finish_and_mark_inactive(void *args) {
-  VALUE self = args;
-  MYSQL_RES *result;
-
+static VALUE disconnect_and_mark_inactive(void *args) {
   GET_CLIENT(self);
```
Comment (burnison) [inline ext/mysql2/client.c]: `self` is no longer defined. In the original patch, the `args` parameter was changed to `VALUE self`.

File: ext/mysql2/client.c [hunk 2]
```diff
@@ -595,20 +595,20 @@ static VALUE do_query(void *args) {
   return Qnil;
 }
 #else
-static VALUE finish_and_mark_inactive(void *args) {
-  VALUE self = args;
-  MYSQL_RES *result;
-
+static VALUE disconnect_and_mark_inactive(void *args) {
```
[the remark this answers is not in the document]
Comment (tamimkh) [reply ext/mysql2/client.c]: Oops. Good catch!

**Comment by tamimkh** [comment]
Closing due to opting for the rebasing with upstream approach.

**Comment by tamimkh** [comment]
Reopening since it looks like our rebase approach wont work due to our old version of rails.

**Pull request closed without merging**